backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala

```diff
@@ -0,0 +1,212 @@
+  
+  when(WaitingForCopyResponses) {
+    case Event(IoSuccess(copyCommand: IoCopyCommand, _), Some(data)) =>
+      val newData = data.remove(copyCommand)
+      if (newData.copyCommandsToWaitFor.isEmpty) succeedAndStop(data.returnCode, data.copiedJobOutputs, data.copiedDetritus)
+      else stay() using Option(newData)
+    case Event(IoFailure(copyCommand: IoCopyCommand, failure), _) =>
+      failAndStop(failure)
+  }
+  
+  whenUnhandled {
+    case Event(AbortJobCommand, _) =>
+      abort()
+    case Event(unexpected, _) =>
+      log.warning(s"Backend cache hit copying actor received an unexpected message: $unexpected in state $stateName")
+      stay()
+  }
+
+  def succeedAndStop(returnCode: Option[Int], copiedJobOutputs: CallOutputs, detritusMap: DetritusMap) = {
+    import cromwell.services.metadata.MetadataService.implicits.MetadataAutoPutter
+    serviceRegistryActor.putMetadata(jobDescriptor.workflowDescriptor.id, Option(jobDescriptor.key), startMetadataKeyValues)
+    context.parent ! JobSucceededResponse(jobDescriptor.key, returnCode, copiedJobOutputs, Option(detritusMap), Seq.empty)
+    context stop self
+    stay()
+  }
+  
+  def failAndStop(failure: Throwable) = {
+    context.parent ! JobFailedNonRetryableResponse(jobDescriptor.key, failure, None)
+    context stop self
+    stay()
+  }
+ 
+  def abort() = {
+    log.warning("{}: Abort not supported during cache hit copying", jobTag)
+    context.parent ! AbortedResponse(jobDescriptor.key)
+    context stop self
+    stay()
+  }
+
+  private def lookupSourceCallRootPath(sourceJobDetritusFiles: Map[String, String]): Path = {
+    sourceJobDetritusFiles.get(JobPaths.CallRootPathKey).map(getPath).get recover {
+      case failure =>
+        throw new RuntimeException(s"${JobPaths.CallRootPathKey} wasn't found for call ${jobDescriptor.call.fullyQualifiedName}", failure)
+    } get
+  }
+
+  /**
+    * Returns a pair of the list of simpletons with copied paths, and copy commands necessary to perform those copies. 
+    */
+  private def processSimpletons(wdlValueSimpletons: Seq[WdlValueSimpleton], sourceCallRootPath: Path): Try[(CallOutputs, Set[PathPair])] = Try {
+    val zero = (List.empty[WdlValueSimpleton], Set.empty[PathPair])
+
+    val (destinationSimpletons, ioCommands) = wdlValueSimpletons.foldLeft(zero)({
+      case ((simpletons, commands), WdlValueSimpleton(key, wdlFile: WdlFile)) =>
+        val sourcePath = getPath(wdlFile.value).get
+        val destinationPath = PathCopier.getDestinationFilePath(sourceCallRootPath, sourcePath, destinationCallRootPath)
+
+        val destinationSimpleton = WdlValueSimpleton(key, WdlFile(destinationPath.pathAsString))
+
+        (simpletons :+ destinationSimpleton, commands + ((sourcePath, destinationPath)))
+      case ((simpletons, commands), nonFileSimpleton) => (simpletons :+ nonFileSimpleton, commands)
+    })
+
+    (WdlValueBuilder.toJobOutputs(jobDescriptor.call.task.outputs, destinationSimpletons), ioCommands)
+  }
```

Comment:
This compiles though tbh I didn't actually run it, but given some recent `foldMap` inspiration:

```
  /**
    * Returns a pair of the list of simpletons with copied paths, and copy commands necessary to perform those copies. 
    */
  private def processSimpletons(wdlValueSimpletons: Seq[WdlValueSimpleton], sourceCallRootPath: Path): Try[(CallOutputs, Set[PathPair])] = Try {
    import cats.instances.list._
    import cats.instances.set._
    import cats.instances.tuple._
    import cats.syntax.foldable._

    val (destinationSimpletons, ioCommands): (List[WdlValueSimpleton], Set[PathPair]) = wdlValueSimpletons.toList.foldMap({
        case WdlValueSimpleton(key, wdlFile: WdlFile) =>
          val sourcePath = getPath(wdlFile.value).get
          val destinationPath = PathCopier.getDestinationFilePath(sourceCallRootPath, sourcePath, destinationCallRootPath)

          val destinationSimpleton = WdlValueSimpleton(key, WdlFile(destinationPath.pathAsString))
          (List(destinationSimpleton), Set((sourcePath, destinationPath)))
        case nonFileSimpleton => (List(nonFileSimpleton), Set.empty)
    })

    (WdlValueBuilder.toJobOutputs(jobDescriptor.call.task.outputs, destinationSimpletons), ioCommands)
  }
```